orchagent/portsorch.cpp

```diff
@@ -530,47 +644,52 @@ void PortsOrch::doPortTask(Consumer &consumer)
                     speed = (uint32_t)stoul(fvValue(i));
             }
 
+            /* Collect information about all received ports */
             if (lane_set.size())
             {
-                /* Determine if the lane combination exists in switch */
-                if (m_portListLaneMap.find(lane_set) !=
-                    m_portListLaneMap.end())
-                {
-                    sai_object_id_t id = m_portListLaneMap[lane_set];
+                m_lanesALiasSpeedMap[lane_set] = make_tuple(alias, speed);
+            }
 
-                    /* Determin if the port has already been initialized before */
-                    if (m_portList.find(alias) != m_portList.end() && m_portList[alias].m_port_id == id)
+            /* Once all ports received, go through the each port and perform appropriate actions:
+             * 1. Remove ports which don't exist anymore
+             * 2. Create new ports
+             * 3. Initialize all ports
+             */
+            if (m_portConfigDone && (m_lanesALiasSpeedMap.size() == m_portCount))
+            {
+                for (auto it = m_portListLaneMap.begin(); it != m_portListLaneMap.end();)
+                {
+                    if (m_lanesALiasSpeedMap.find(it->first) == m_lanesALiasSpeedMap.end())
                     {
-                        SWSS_LOG_INFO("Port has already been initialized before alias:%s", alias.c_str());
+                        if (!removePort(it->first))
```

Comment:
Maybe it's better to redefine the input parameter of removePort. The function removePort can use port object id directly.